src/bindings/js/node/src/infer_request.cpp

```diff
@@ -49,24 +49,37 @@ Napi::Value InferRequestWrap::set_input_tensor(const Napi::CallbackInfo& info) {
 Napi::Value InferRequestWrap::infer(const Napi::CallbackInfo& info) {
     if (info.Length() == 0)
         _infer_request.infer();
-    else if (info.Length() == 1 && info[0].IsObject()) {
-        const auto inputs = info[0].As<Napi::Object>();
-        auto keys = inputs.GetPropertyNames();
+    // TO_DO single TensorWrap object
+    // TO_DO TypedArray with Tensor data
+    else if (info.Length() == 1 && info[0].IsArray()) {
+        infer(info[0].As<Napi::Array>());
+    } else if (info.Length() == 1 && info[0].IsObject()) {
+        infer(info[0].As<Napi::Object>());
+    } else {
+        reportError(info.Env(), "Infer method takes as an argument an array or an object with tensors.");
+    }
+    return Napi::Value();
+}
+
+void InferRequestWrap::infer(const Napi::Array& inputs) {
+    for (size_t i = 0; i < inputs.Length(); i++) {
```

Comment:
```suggestion
    for (size_t i = 0; i < inputs.Length(); ++i) {
```
If want just increment use pre-increment, is good practice. For fundamental type id doesn't matter will be well optimized.